File: Agentic-AI-Travel-Advisor/shared/Core/Entities/Room.cs
namespace TravelAdvisor.Core.Entities;

public class Room
{
    public int Id { get; set; }
    public int HotelId { get; set; }
    public string Name { get; set; } = string.Empty;
    public string RoomType { get; set; } = string.Empty;
    public decimal PricePerNight { get; set; }
    public int Capacity { get; set; }
    public bool IsAvailable { get; set; } = true;

    public Hotel Hotel { get; set; } = null!;
    public ICollection<Booking> Bookings { get; set; } = [];
}

File: Agentic-AI-Travel-Advisor/mobile-app/lib/models/room.dart
class Room {
  final int id;
  final int hotelId;
  final String name;
  final String roomType;
  final double pricePerNight;
  final int capacity;
  final bool isAvailable;

  const Room({
    required this.id,
    required this.hotelId,
    required this.name,
    required this.roomType,
    required this.pricePerNight,
    required this.capacity,
    required this.isAvailable,
  });

  factory Room.fromJson(Map<String, dynamic> json) {
    return Room(
      id: json['id'] as int,
      hotelId: json['hotelId'] as int,
      name: json['name'] as String,
      roomType: json['roomType'] as String,
      pricePerNight: (json['pricePerNight'] as num).toDouble(),
      capacity: json['capacity'] as int,
      isAvailable: json['isAvailable'] as bool,
    );
  }
}
